classes/task/PKPUsageStatsLoader.php

```diff
@@ -114,88 +105,31 @@ protected function executeActions(): bool
-        }
-        return ($processFilesResult && !$processingDirError);
-    }
-
-    /**
-     * Delete entries in usage stats temporary tables by loadId
-     */
-    protected function deleteByLoadId(string $loadId): void
-    {
-        $this->temporaryInstitutionsDao->deleteByLoadId($loadId);
-        $this->temporaryTotalsDao->deleteByLoadId($loadId);
-        $this->temporaryItemInvestigationsDao->deleteByLoadId($loadId);
-        $this->temporaryItemRequestsDao->deleteByLoadId($loadId);
-    }
 
-    /**
-     * Insert usage stats log entry into temporary tables
-     */
-    protected function insertTemporaryUsageStatsData(object $entry, int $lineNumber, string $loadId): void
-    {
-        try {
-            $this->temporaryTotalsDao->insert($entry, $lineNumber, $loadId);
-            $this->temporaryInstitutionsDao->insert($entry->institutionIds, $lineNumber, $loadId);
-            if (!empty($entry->submissionId)) {
-                $this->temporaryItemInvestigationsDao->insert($entry, $lineNumber, $loadId);
-                if ($entry->assocType == Application::ASSOC_TYPE_SUBMISSION_FILE) {
-                    $this->temporaryItemRequestsDao->insert($entry, $lineNumber, $loadId);
-                }
+        if ($processFilesResult) {
+            $site = Application::get()->getRequest()->getSite();
+            $jobs = [];
+            foreach ($this->logFiles as $filePath) {
+                $jobsPerFile = $this->getFileJobs($filePath, $site);
+                $jobs = array_merge($jobs, $jobsPerFile);
             }
-        } catch (\Illuminate\Database\QueryException $e) {
-            $this->addExecutionLogEntry(__('admin.scheduledTask.usageStatsLoader.insertError', ['file' => $loadId, 'lineNumber' => $lineNumber, 'msg' => $e->getMessage()]), ScheduledTaskHelper::SCHEDULED_TASK_MESSAGE_TYPE_ERROR);
-        }
-    }
-
-    /**
-     * Get valid assoc types that an usage event can contain
-     */
-    abstract protected function getValidAssocTypes(): array;
-
-    /**
-     * Validate the usage stats log entry
-     *
-     * @throws Exception.
-     */
-    protected function isLogEntryValid(object $entry): void
-    {
-        if (!$this->validateDate($entry->time)) {
-            throw new Exception(__('admin.scheduledTask.usageStatsLoader.invalidLogEntry.time'));
-        }
-        // check hashed IP ?
-        // check canonicalUrl ?
-        if (!is_int($entry->contextId)) {
-            throw new Exception(__('admin.scheduledTask.usageStatsLoader.invalidLogEntry.contextId'));
-        }
-        if (!empty($entry->submissionId) && !is_int($entry->submissionId)) {
-            throw new Exception(__('admin.scheduledTask.usageStatsLoader.invalidLogEntry.submissionId'));
-        }
+            foreach ($this->months as $month) {
+                $compileMonthlyMetricsJob = new CompileMonthlyMetrics($month, $site);
+                $jobs = array_merge($jobs, [$compileMonthlyMetricsJob]);
```

Comment:
Is it possible for the list of files to grow into an unmanageable size?
For example, if the user decides to reprocess his whole archive.
If yes, then it could be interesting to add a limit.

Reply:
The CLI tool for re-processing takes 'only' one month.
If the user keeps the daily stats, it would be possible to copy all archived log files into the usageEventLogs folder and run the scheduled task -- then as you said all would be considered...
I hope the admins know what are they doing :sweat_smile: 
Hmmm... If we would add limit for 10-30 files, the admin could however ignore it and start a new scheduled task... So maybe first we will need to solve the two cases from the beginning.
I will document it all in a new issue...

Reply:
I think this needs to be added to the https://docs.pkp.sfu.ca/admin-guide/en/statistics#reprocess-log-files.